safehttp/plugins/xsrf/xsrf.go

```diff
@@ -34,24 +42,47 @@ type UserIdentifier interface {
 }
 
 // Interceptor implements XSRF protection. It requires an application key and a
-// storage service. The appKey uniquely identifies each registered service and
+// storage service. The AppKey uniquely identifies each registered service and
 // should have high entropy. The storage service supports retrieving ID's of the
-// application's users. Both the appKey and user ID are used in the XSRF
-// token generation algorithm.
+// application's users. Both the AppKey and user ID are used in the XSRF
+// token generation and verification algorithm.
 type Interceptor struct {
 	AppKey     string
 	Identifier UserIdentifier
 }
 
+type tokenCtxKey struct{}
+
+// Token returns the XSRF token from the safehttp.IncomingRequest Context and a
+// nil error, if present, and, otherwise, returns an empty string and a non-nil
+// error.
```

Comment:
"Token extract the XSRF token from the incoming request. If it is not present, returns a non-nil error." is enough.

- no need to write `safehttp.IncomingRequest` or that it gets the token from it's `Context`
- in Go, it is the convention that nil error indicates success, so no need to write about it
- in Go, you usually assume that if one of the return values is a non-nil error, you cannot depend on other return values

In general, try to think about comments in a more declarative, than imperative way. For instance, imagine a function `func MinInt(a, b int) int`. You have two ways of commenting:

1. "MinInt takes two integers and if the first one is smaller or equal to the second one, it returns the first. If the second one is smaller, it returns the second."
2. "MinInt takes two integers and returns the smaller one."